group-coordinator/src/main/java/org/apache/kafka/coordinator/group/share/ShareCoordinatorShard.java

```diff
@@ -153,7 +171,34 @@ public void onUnloaded() {
 
   @Override
   public void replay(long offset, long producerId, short producerEpoch, Record record) throws RuntimeException {
+    ApiMessageAndVersion key = record.key();
+    ApiMessageAndVersion value = record.value();
+
+    switch (key.version()) {
+      case 0: // ShareSnapshot
```

Comment:
It's always struck me as a little strange that there are no constant definitions for these values. I think that would be a useful idea to introduce with the share coordinator. Possibly in the ShareCoordinator class.